File: backend/schemas/application.py
from datetime import datetime
from enum import StrEnum
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from schemas.job import JobResponse


class ApplicationStatus(StrEnum):
    QUEUED = "queued"
    PENDING_REVIEW = "pending_review"
    IN_PROGRESS = "in_progress"
    APPLIED = "applied"
    FAILED = "failed"
    SKIPPED = "skipped"
    WITHDRAWN = "withdrawn"


class ApplicationDetail(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    user_id: UUID
    job_id: UUID | None = None
    status: ApplicationStatus
    applied_at: datetime | None = None
    resume_used_url: str | None = None
    cover_letter_used: str | None = None
    screenshot_url: str | None = None
    error_message: str | None = None
    created_at: datetime
    job: JobResponse | None = None


class ApplicationListResponse(BaseModel):
    applications: list[ApplicationDetail]
    total: int
    page: int
    per_page: int
    pages: int


class ApplicationStats(BaseModel):
    total: int
    applied: int
    pending: int
    failed: int
    skipped: int
    this_week: int
    success_rate: float  # applied / (applied + failed) * 100

File: frontend/src-v1-deprecated/types/application.ts
import type { JobResponse } from "./job";

export type ApplicationStatus =
  | "queued"
  | "pending_review"
  | "in_progress"
  | "applied"
  | "failed"
  | "skipped"
  | "withdrawn";

export interface ApplicationDetail {
  id: string;
  user_id: string;
  job_id?: string | null;
  status: ApplicationStatus;
  applied_at?: string | null;
  resume_used_url?: string | null;
  cover_letter_used?: string | null;
  screenshot_url?: string | null;
  error_message?: string | null;
  created_at: string;
  job?: JobResponse | null;
}

export interface ApplicationListResponse {
  applications: ApplicationDetail[];
  total: number;
  page: number;
  per_page: number;
  pages: number;
}

export interface ApplicationStats {
  total: number;
  applied: number;
  pending: number;
  failed: number;
  skipped: number;
  this_week: number;
  success_rate: number;
}
